drivers/SmartThings/zigbee-window-treatment/src/screen-innovations/init.lua

```diff
@@ -0,0 +1,174 @@
+    elseif level == 100 then
+      event = capabilities.windowShade.windowShade.closed() --Bug #16054
+    else
+      event = capabilities.windowShade.windowShade.partially_open()
+    end
+  end
+
+  -- update status
+  if event ~= nil then
+    device:emit_event(event)
+  end
+
+  -- update level
+  device:emit_event(capabilities.windowShadeLevel.shadeLevel(level))
+end
+
+-- this is motor running_direction_attr_handler
+local function running_direction_attr_handler(driver, device, value, zb_rx)
+  local status = value.value
+  if status == 1 then
+    running_direction = motor_states.OPENING
+  elseif status == 2 then
+    running_direction = motor_states.CLOSING
+  else
+    running_direction = motor_states.IDLE
+  end
+end
+
+-- this is do_configure
+local function do_configure(self, device)
+  -- configure elements
+  device:configure()
+  device:send(device_management.build_bind_request(device, WindowCovering.ID, self.environment_info.hub_zigbee_eui))
+  device:send(WindowCovering.attributes.CurrentPositionLiftPercentage:configure_reporting(device, 1, 3600, 1))
+  device:send(device_management.build_bind_request(device, PowerConfiguration.ID, self.environment_info.hub_zigbee_eui))
+  device:send(PowerConfiguration.attributes.BatteryPercentageRemaining:configure_reporting(device, 1, 3600, 1))
+  device:send(device_management.build_bind_request(device, Basic.ID, self.environment_info.hub_zigbee_eui))
+  device:send(Basic.attributes.PowerSource:configure_reporting(device, 1, 3600))
+
+  -- read elements
+  device.thread:call_with_delay(3, function(d)
+    device:send(Basic.attributes.ApplicationVersion:read(device))
+    update_device_info(device)
+  end)
+end
+
+-- this is do_refresh
+local do_refresh = function(self, device)
+  update_device_info(device)
+end
+
+-- this is battery_perc_attr_handler
+local function battery_perc_attr_handler(driver, device, value, zb_rx)
+  local converted_value = value.value / 2
+  converted_value = utils.round(converted_value)
+  -- update battery percentage only motor is in idle state --Bug #16055
+  if running_direction == motor_states.IDLE then
+    device:emit_event_for_endpoint(zb_rx.address_header.src_endpoint.value,
+      capabilities.battery.battery(utils.clamp_value(converted_value, 0, 100)))
+  end
+end
+
+-- create the handler object
+local screeninnovations_roller_shade_handler = {
+  NAME = "screeninnovations_roller_shade_handler",
+  lifecycle_handlers = {
+    added = device_added,
+    doConfigure = do_configure
+  },
+  capability_handlers = {
+    [capabilities.windowShadeLevel.ID] = {
+      [capabilities.windowShadeLevel.commands.setShadeLevel.NAME] = window_shade_level_cmd
+    },
+    [capabilities.windowShadePreset.ID] = {
+      [capabilities.windowShadePreset.commands.presetPosition.NAME] = window_shade_preset_cmd
+    },
+    [capabilities.refresh.ID] = {
+      [capabilities.refresh.commands.refresh.NAME] = do_refresh,
```

Comment:
you could just directly reference `update_device_info` here instead of using a wrapper function

Reply:
Removed the wrapper function.